### PR TITLE
Fix TensorFlowScorer duplicate parameter name Model

### LINKED ISSUE #1039: TensorFlowScorer has duplicate parameter Model
ML.NET version 0.6.0

See [manifest.json](https://github.com/dotnet/machinelearning/blob/master/test/BaselineOutput/Common/EntryPoints/core_manifest.json) 

One Model is Input param, another Model is output param. This causes issues in auto code generators that automatically create method signatures based on manifest. Pls use model_location for input param

### PR DESCRIPTION
**Pull request by ganik**
* Fix TensorFlowScorer duplicate parameter name Model
* Add unit test to detect duplicate parameter names in entrypoints
fixes #1039 

New unit test detected duplication of parameter names in following components:
Data.DataViewReference
Models.CrossValidator
Models.CrossValidationResultsCombiner
Models.PipelineSweeper
Models.PipelineSweeper
Models.SweepResultExtractor
Models.TrainTestEvaluator
Transforms.TwoHeterogeneousModelCombiner
Transforms.ManyHeterogeneousModelCombiner

I would like to have a separate PRs to fix these entrypoints. The new unit test will prevent further duplication from now on.

### CODE REVIEW

File: test/Microsoft.ML.Core.Tests/UnitTests/TestEntryPoints.cs [hunk 1]
```diff
@@ -300,6 +300,44 @@ public void EntryPointCatalog()
             Done();
         }
 
+        [Fact]
+        public void EntryPointCatalogCheckDuplicateParams()
+        {
+            // Run this test to prevent introducing duplicate param names in entrypoints
+            // TODO: fix entrypoints in excludeSet from having duplicate param names
+            var excludeSet = new HashSet<string>();
```
Comment (abgoswam) [inline test/Microsoft.ML.Core.Tests/UnitTests/TestEntryPoints.cs]: >excludeSet [](start = 16, length = 10)

Should we fix these as well as part of this PR ?  

Or is there a separate issue created to handle these separately ?

**Comment by Ivanidzo4ka** [comment]
Out of curiosity, why you think it's a bug to have two same names in under different subnodes?
One Model is Input node, one is in output.  And do you care about only this case, or for example i can have subnode A and subnode B in input node and both would have property Model, would it also count as bug?

File: test/Microsoft.ML.Core.Tests/UnitTests/TestEntryPoints.cs (hunk 1 above)
Comment (ganik) [reply test/Microsoft.ML.Core.Tests/UnitTests/TestEntryPoints.cs]: Pls see the description of the PR where I suggest to fix each component in separate PR. Firstly this will keep PRs small and focused on specific component. Secondly to get consensus on name need to talk to original author. I ll open issues on each of it.

Comment (ganik) [reply test/Microsoft.ML.Core.Tests/UnitTests/TestEntryPoints.cs]: I prefer to fix each component in separate PR. Firstly this will keep PRs small and focused on specific component. Secondly to get consensus on name one needs to talk to original author. I can open issues on each of it. 

---
In reply to: [223161784](https://github.com/dotnet/machinelearning/pull/1169#discussion_r223161784) [](ancestors = 223161784)

**Comment by ganik** [comment]
This is not a bug per se. This is about how to make entrypoint API easy to use by 3rd parties/clients. Since I have been implementing such a client, I can share my experience. We have viewed "entrypoint" as a function, sort of RPC that you can call. All the parameters for "entrypoint" are parameters of such a function. Even though Model is set as "Output" param, its actually required to set this param to a filepath so ML.NET can output zipped model there. Therefore entrypoint needs 2 model params: one to specify filepath for input model, another one to specify path for output model. I am not sure if I know any programming language where function can accept parameters with the same name. 
Of course its possible to view "entrypoint" not as a method but some sort of class, or break up params into separate Input and Output structs and populate separately, but in the end, on a highest level, what is going to surface to user is a function with params to specify input and output model paths. It would be easier to avoid intermediates and being able to have that high level function built right on top of manifest.json specs. 

---
In reply to: [427531879](https://github.com/dotnet/machinelearning/pull/1169#issuecomment-427531879) [](ancestors = 427531879)

**Comment by ganik** [comment]
@codemzs is added to the review. #Closed
